### DISCUSSION TITLE
Solis Cloud API document v2.0

### DISCUSSION THREAD

**Discussion by mtwbkr** [ZuinigeRijder/SolisCloud2PVOutput#31] [Q&A] [answered]
Hi, I noticed API document v2.0 is available when API access was recently granted to my Solis Cloud account.

I have never used code before but am very interested to get my Solis inverter data into PVOutput. I have been trying to practice with the script in Python Anywhere and have successfully updated the config details required. Running the script gives the following error:

ERROR: https://www.soliscloud.com:13333/v1/api/userStationList -> Tunnel connection failed: 403 Forbidden

Can you provide any pointers to fix this issue?

Thanks

[SolisCloud Platform API Document V2.0.pdf](https://github.com/ZuinigeRijder/SolisCloud2PVOutput/files/14878787/SolisCloud.Platform.API.Document.V2.0.pdf)

**Comment by PeJeWe** [comment]
Maybe you have to open that port in your router.

**Comment by ZuinigeRijder** [comment]
Which Operating System are you running from? Which Python version?

Some pointers of possible causes:
[Do you run in a proxy](https://stackoverflow.com/questions/71138889/max-retries-exceeded-caused-by-proxyerrorcannot-connect-to-proxy-oserror) (so you do not have direct internet connection)?
[Firewall?](https://wetopi.com/error-code-403-forbidden/)
[Do you use pythonanyware?](https://help.pythonanywhere.com/pages/403ForbiddenError/)

**Reply by mtwbkr** [reply, marked as answer]
Thank you. I was test trialling using a free account on Python Anywhere - it uses a proxy and Solis Cloud is not on the list of accessible sites. Paid versions do not have the limitation.
I have ordered a Raspberry Pi 4 and will start my journey in this new world.

**Comment by ZuinigeRijder** [comment]
See here some information of [configuring a proxy using environment variables](https://docs.python-requests.org/en/latest/user/advanced/#proxies) or [see this page](https://stackoverflow.com/questions/11726881/how-to-set-an-http-proxy-in-python-2-7) if this is the problem.
